CoastalResilienceNetwork/geosite-framework-build
# Build static site

## Overview

Updates the tool to be able to build the static site. 

This currently works on OSX, and requires some changes to work on other OSs. 

Todo:
- [x] ~~Make sure [python dependencies](https://github.com/CoastalResilienceNetwork/GeositeFramework/blob/feature/task-1/src/GeositeFramework/requirements.txt) are installed for the framework.~~ Moved to https://github.com/CoastalResilienceNetwork/geosite-framework-build/issues/41
- [x] Build on Windows (Fixed by https://github.com/CoastalResilienceNetwork/GeositeFramework/pull/1138/commits/7786ffa2da10bc5603e1381feffda949eb8db586)
- [x] Build on Linux. Will need help from @mmcfarland on this.
- [x] Verify errors in framework build scripts bubble up to build tool output.
- [x] Do some more testing to ensure that arguments like `--prod` work as expected.
- [ ] Update paths after #1100 is implemented.

Connects #37

### Demo

```
python build.py gulfmex-region --framework-branch cpc/build-fixes --region-branch development
Cloning gulfmex-region_config@HEAD...
Attempting to checkout the development branch
Branch 'development' set up to track remote branch 'development' from 'origin' by rebasing.
Cloning GeositeFramework@HEAD...
Attempting to checkout the cpc/build-fixes branch
Branch 'cpc/build-fixes' set up to track remote branch 'cpc/build-fixes' from 'origin' by rebasing.
Cloning GeositeFramework/src/GeositeFramework/plugins/future_habitat@HEAD...
Attempting to checkout the development branch
Cloning GeositeFramework/src/GeositeFramework/plugins/eca@HEAD...
Attempting to checkout the development branch
Branch 'development' set up to track remote branch 'development' from 'origin' by rebasing.
Cloning GeositeFramework/src/GeositeFramework/plugins/regional-planning@HEAD...
Attempting to checkout the development branch
Cloning GeositeFramework/src/GeositeFramework/plugins/community_planning@HEAD...
Attempting to checkout the development branch
Cloning GeositeFramework/src/GeositeFramework/plugins/flood_slr@HEAD...
Attempting to checkout the development branch
Cloning GeositeFramework/src/GeositeFramework/plugins/gulf-mex-crs@HEAD...
Attempting to checkout the development branch
Cloning GeositeFramework/src/GeositeFramework/plugins/habitat_explorer@HEAD...
Attempting to checkout the development branch
Cloning GeositeFramework/src/GeositeFramework/plugins/risk_explorer@HEAD...
Attempting to checkout the development branch
Cloning GeositeFramework/src/GeositeFramework/plugins/restoration_explorer@HEAD...
Attempting to checkout the development branch
Copying region.json...
Copying partners.html...
[Errno 2] No such file or directory: 'partners.html'
Copying Proxy.config...
[Errno 2] No such file or directory: 'Proxy.config'
Copying plugins...
Copying img...
Copying Views...
Copying methods...
Copying sims...
cannot copy tree 'sims': not a directory
Copying xml...
Copying docs...
Copying locales...
cannot copy tree 'locales': not a directory

---------------------------------------
CoastalResilienceNetwork/gulfmex-region was built successfully
```

## Testing Instructions

- Run the following command to build a region instance:

```
python build.py <region-name> --framework-branch feature/task-1 --region-branch development
```
- Go to the `output` directory and unzip the region zip.
- Start a web server in the source directory. **Note** do not use `scripts/server.py`. This will rebuild the site. Use `python -m SimpleHTTPServer` or similar directly.
- Visit the site in the browser, and verify it works.

- Run the following command to test that errors in the region will bubble up and prevent the site from building:

```
python build.py nj-spanish-region caseycesari --framework-branch feature/task-1 --region-branch development
```
